### DISCUSSION TITLE
Advice on Using MONAI for Carina Localization

### DISCUSSION THREAD

**Discussion by robertlario** [Project-MONAI/MONAILabel#1743] [General]
Hi,

I’m looking for advice on how to apply MONAI for my use case. Specifically, I would like to send a medical image of a person's torso to a web service and receive the XY coordinates (JSON object) of the Carina's location in the image. Additionally, I’d like the ability to provide feedback to the service with the correct XY coordinates for training and improvement purposes.

Is there an example in MONAI or a similar setup that aligns with this idea?

Thanks in advance for your help and suggestions.

Best regards,

Robert

**Comment by diazandr3s** [comment]
Hi @robertlario,

This is a great idea. Is this 2D (X-Rays) or 3D images (CT or MR volumes)?
From what I understand, this is a regression task - You input an image and the model ouputs coordinates.

Here is an example of how you can train a network for a regression task on 3D images: https://github.com/Project-MONAI/tutorials/tree/main/3d_regression

You could take advantage of the REST APIs offered by MONAI Label to serve this model. If it is too much, you could create your own APIs using FastAPI to serve the model.

Hope this helps,

**Comment by robertlario** [comment]
When Resized transformation is applied to an image does the model report the localized coordinate based upon the Resized dimensions or the images original dimensions? I would assume the Resized dimensions. Given this is the case, is there an easy utility to convert the model reported coordinates back into the original images dimension?

Thank you

**Reply by diazandr3s** [reply]
Hi @robertlario,

I'm sorry for the late reply.

The Resize transform has an invert method to return the image to the original size: https://github.com/Project-MONAI/MONAI/blob/46a5272196a6c2590ca2589029eed8e4d56ff008/monai/transforms/spatial/array.py#L853-L854

You could use of this method.

Hope this helps,
